realm/realm-library/src/androidTest/java/io/realm/rule/TestRealmConfigurationFactory.java

```diff
@@ -94,6 +94,18 @@ public RealmConfiguration createConfiguration() {
         return configuration;
     }
 
+    public RealmConfiguration createConfiguration(String subDir, String name) {
+        final File folder = new File(getRoot(), subDir);
+        //noinspection ResultOfMethodCallIgnored
+        folder.mkdirs();
```

Comment:
Fore the exceptional case where it might fail, perhaps `assertTrue(folder.mkDirs())`. You could also remove the noinspection then